ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -97,12 +99,14 @@ public static String hostname() {
   public void run() {
     LOG.info("Starting Worker thread");
     boolean computeStats = conf.getBoolVar(HiveConf.ConfVars.HIVE_MR_COMPACTOR_GATHER_STATS);
+    boolean metricsEnabled = conf.getBoolVar(HiveConf.ConfVars.HIVE_SERVER2_METRICS_ENABLED);
```

Comment:
I didn't see anywhere in HS2 to check this property before using perflogger, the metricsfactory will be disabled globally if this is not enabled. You can remove it.

Reply:
fixed